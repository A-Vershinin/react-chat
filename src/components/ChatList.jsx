import React, { Component } from 'react';
import classNames from 'classnames';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import Paper from '@material-ui/core/Paper';
import Avatar from 'material-ui/Avatar';

import ChatItem from './ChatItem.jsx';
import titleInitials from '../utils/title-initials';

const styles = theme => ({
  chatLayout: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    overflow: 'hidden',
  },
  messagesWrapper: {
    overflowX: 'scroll',
    height: '100%',
    width: '100%',
    paddingTop: theme.spacing.unit * 3,
    paddingBottom: '120px',
  },
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrappperFromMe: {
    justifyContent: 'flex-end',
  },
});

class ChatList extends Component {
  render() {
    const { classes, messages } = this.props;

    return (
      <div className={classes.messagesWrapper}>
      {messages && messages.map((message, index) => {
        const isMessageFromMe = message.sender === 'me';

        const userAvatar = (
          <Avatar> {titleInitials(message.sender[0])} </Avatar>
        );

        return (
          <div key={index} className={classNames(classes.messageWrapper,
            isMessageFromMe && classes.messageWrappperFromMe)}>
            <ChatItem
              isMessageFromMe={isMessageFromMe}
              userAvatar={userAvatar}
              message={message}
            />
        </div>
        );
      })}
      </div>
    );
  }

}

export default withStyles(styles)(ChatList);
