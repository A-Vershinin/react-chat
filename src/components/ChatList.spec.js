/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatList from './ChatItem.jsx';

jest.mock('./ChatItem', () => () => 'ChatItem');

const mockProps = {
  classes: {
    chatsList: 'lorem lorem',
    noChats: 'lorem lorem abc',
  },
  chats: [
    {
      createdAt: '2018-03-16T10:53:23.200Z',
      creator: {
        firstName: 'firstName',
        lastName: 'lastName',
        username: 'username',
        _id: '12345',
      },
      members: [
        {
          firstName: 'firstName',
          lastName: 'lastName',
          username: 'username',
          _id: '12345',
        },
      ],
      title: 'My Chat',
      updatedAt: '2018-03-16T10:53:23.200Z',
    },
  ],
  activeChat: {
    _id: '12345',
    title: 'My Chat',
    createdAt: '2018-03-16T10:53:23.200Z',
    updatedAt: '2018-03-16T10:53:23.200Z',
    creator: {
      firstName: 'firstName',
      lastName: 'lastName',
      username: 'username',
      _id: '12345',
    },
    members: [
      {
        firstName: 'firstName',
        lastName: 'lastName',
        username: 'username',
        _id: '12345',
      },
    ],
  },
  disabled: false,
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatList {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatList {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when ChatList disabled', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatList {...mockProps} disabled />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
