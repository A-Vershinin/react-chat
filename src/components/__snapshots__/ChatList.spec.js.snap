// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `"ChatItem"`;

exports[`<ChatList /> renders when ChatList disabled 1`] = `"ChatItem"`;
